src/middlewared/middlewared/plugins/cluster_linux/ctdb_ips.py

```diff
@@ -33,12 +33,16 @@ async def common_validation(self, data, schema_name, verrors):
 
         verrors.check()
 
+        if schema_name == 'public_delete':
+            return
+
         if schema_name in ('private_create', 'public_create'):
             existing_public_ips = {}
 
             if schema_name == 'public_create':
+                pnn = data.get('pnn') or await self.middleware.call('ctdb.general.pnn')
```

Comment:
`pnn = data.get('pnn') or await self.middleware.call('ctdb.general.pnn')` can be called once above the `if schema_name in ('private_create', 'public_create'):` line I think